### PR TITLE
[3672] Lead school welcome email

### PR DESCRIPTION
**Pull request by mooktakim**
### Context

https://trello.com/c/C40nFwLy/3672-lead-school-welcome-email

### Changes proposed in this pull request

* Updated existing `WelcomeEmailMailer` to use new lead school user welcome template
* Updated `SendWelcomeEmailService` to only send email to lead school users

### Guidance to review

* 🤷‍♀️

### Important business

* Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
* Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?

### CODE REVIEW

**Comment by mooktakim** [comment]
[Lead school welcome email](https://trello.com/c/C40nFwLy/3672-lead-school-welcome-email)

**Comment by mooktakim** [comment]
[Lead school welcome email](https://trello.com/c/C40nFwLy/3672-lead-school-welcome-email)

File: app/services/send_welcome_email_service.rb [hunk 1]
```diff
@@ -10,6 +10,7 @@ def initialize(current_user:)
   def call
     return unless FeatureService.enabled?(:send_emails)
     return if current_user.welcome_email_sent_at
+    return unless current_user.lead_school?
```
Comment (AbigailMcP) [inline app/services/send_welcome_email_service.rb:13]: I have a hunch that `current_user.lead_school?` will return `nil` for _all_ users at this point.
The organisation 'choosing' page is shown after the user signs in, so there'll be no `session[:current_organisation]`.

See: https://github.com/DFE-Digital/register-trainee-teachers/blob/main/app/lib/user_with_organisation_context.rb#L29

**Review by AbigailMcP — APPROVED** [review]
👍

**Pull request merged**